portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -38,17 +38,27 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    // Debug
+    // response.getWriter().println(request);
+    // response.getWriter().println(request.getParameter("quantity"));
+
+    
     Query query = new Query("Comment").addSort("postTime", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
     List<Comment> comments = new ArrayList<>();
+    int count = 0;
+    int quantity = Integer.parseInt(request.getParameter("quantity"));
     for (Entity entity : results.asIterable()) {
+      if (count >= quantity) break;
```

Comment:
nit:  style-wise I would suggest moving break to the next line.  Curly brackets are optional, although I usually prefer adding them in even for one-liners like this.  (Google style guide for C++ also recommends adding optional brackets).